src/main/java/com/kodekonveyor/cdd/run/impl/ContractRunnerServiceImpl.java

```diff
@@ -195,16 +198,13 @@ private void checkReturnValue(
       final Description description
   ) {
     if (returnValue == null) {
```

Comment:
```
if (returnValue != null ? !returnValue.equals(answer) : answer != null) {
    notifyFailureForBadResult(
        contract, notifier, description, answer, returnValue
    );
}
```
And the condition can be extracted to method.

Reply:
I think the `predicate? case : othercase` syntax is confusing. I would never use it.
Indeed I propose to put it to the list of banned constructs.
Turning the ifs to a boolean expression does not improve readability, code complexity or execution speed.
I also do not think that the complexity or readability of the current code would warrant to refactor it to another function. Actually it would result in the original function containing an if and the notifyFailureForBadResult call, and the new function being the same as this one, just the notifyFailureForBadResult replaced with a return. Two functions, both nearly as complex as this one.

Reply:
What about Optional?
if (!Optional.ofNullable(returnValue).equals(Optional.ofNullable(answer)))
Nullable values could be converted to Optional as early as possible IMHO

I wouldn't say ternary conditional operator should be banned, since it reduces LLOC and improves readability in many times.

E.g.
`
int max = (a > b) ? a : b;
`
would be
`
int max;
if (a > b) {
    max = a;
} else {
    max = b;
}
`
without ternatry operator, that is too verbose for me compared to the simplicity of the task.